pkg/controller/nodes/task/handler.go

```diff
@@ -561,6 +561,20 @@ func (t Handler) Handle(ctx context.Context, nCtx handler.NodeExecutionContext)
 	// So now we will derive this from the plugin phase
 	// TODO @kumare re-evaluate this decision
 
+	tk, err := tCtx.tr.Read(ctx)
+	if err != nil {
+		logger.Errorf(ctx, "Failed to read TaskTemplate, error :%s", err.Error())
+		return handler.UnknownTransition, err
+	}
+	inputs := &core.LiteralMap{}
```

Comment:
Can we only update the methods that access the input reader instead of reading inputs here?
e.g. Update CheckCatalogCache method and update ToTaskExecutionEvent method... etc. to read the inputs before building the event?